.github/workflows/test-changed-firestore.yml

```diff
@@ -28,27 +34,115 @@ jobs:
           cp config/ci.config.json config/project.json
           yarn
       - name: build
-        run: yarn build:changed firestore
-      - name: Run tests if firestore or its dependencies has changed
-        run: yarn test:changed firestore
+        id: build
+        run: yarn build:changed firestore | egrep "Skipping all"
```

Comment:
What happens if there's a build failure that doesn't produce the output Skipping All. Does it still generate an error in `steps.build.outcome` (which is tested below)?

Reply:
Yes, the tests will run and they will fail for missing dependencies because build failed.

Reply:
So if there's a build failure then `steps.build.outcome` will be `success`?

Is there a way that we can make this job fail instead, so that the other jobs don't even start? Seems like it would be more clear as to where the problem is when an error occurs.

Reply:
Also, by piping the output from `yarn build:changed firestore` through `egrep` the entire output is lost. So if the build fails there will be no logs indicating _why_ the build failed. Please find a way to do this while preserving the command's output in the logs.

Reply:
You'd probably want to redirect stderr as well, so that it shows up inline. Using the `tee` command seems like a good candidate here:

```
yarn build:changed 2>&1 | tee ${{ runner.temp }}/yarn.log.txt
```

This would print the output to stdout and _also_ save it into a temp file. The next line in the step's bash script could run `egrep`.